packages/sit-onyx/src/styles/required.scss

```diff
@@ -1,28 +1,39 @@
-// `onyx-required-marker` and `onyx-optional-marker` is used internally to populate the the "::after" pseudo element with the required indicator.
-// the user can set `onyx-use-optional` to switch from the default "required" indicator to the "optional" indicators.
-.onyx-required-marker {
-  &::after {
-    font-family: var(--onyx-font-family);
-    content: "*";
-    color: var(--onyx-color-text-icons-danger-intense);
-    padding-left: var(--onyx-spacing-5xs);
-  }
+// `onyx-required-marker` is used internally to populate the the "::after" pseudo element with the required marker.
+// the user can set `onyx-use-optional` and `onyx-use-required` to switch between the indicators.
 
-  .onyx-use-optional &::after {
-    // don't show required marker in optional mode
-    all: unset;
-  }
+:root {
+  // We use a trick to toggle values using custom properties https://lea.verou.me/blog/2020/10/the-var-space-hack-to-toggle-multiple-values-with-one-custom-property/
+  // This allows us to make use of custom property inheritance
+  --ON: initial;
+  --OFF: ;
+}
+
+/* default */
+:root,
+.onyx-use-required {
+  --onyx-use-required: var(--ON);
+  --onyx-use-optional: var(--OFF);
+}
+
+.onyx-use-optional {
+  --onyx-use-required: var(--OFF);
+  --onyx-use-optional: var(--ON);
+}
+
+.onyx-required-marker::after {
+  font-family: var(--onyx-use-required, var(--onyx-font-family));
```

Comment:
Why does this work? When required is ON, the fallback value would not be used because its value is "initial".

Reply:
The trick is that the value of `initial` for a custom property will trigger the fallback, while a whitespace will not.

Reply:
I added a link to an explanation above